pkg/api/steve/supportconfigs/supportconfigs.go

```diff
@@ -0,0 +1,57 @@
+package supportconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/version"
+)
+
+// supportConfig represents the serializable struct of support related information common to all rancher setups
+type supportConfig struct {
+	// SupportEligible - if this rancher product is eligible for support.
+	SupportEligible bool `json:"support_eligible,omitempty"`
+	// Platform - the platform (arm, x86_64, etc) rancher is running on. As of now, always x86
+	Platform string `json:"platform"`
+	// Product - suse product string identifying rancher - see getRancherProductString
+	Product string `json:"product"`
+}
+
+// supportConfigHandler contains the necessary clients/controllers for retrieving a support config
+type supportConfigHandler struct {
+}
+
+// getSupportConfig retrieves a supportConfig
+func (h *supportConfigHandler) getSupportConfig() supportConfig {
+	return supportConfig{
+		SupportEligible: false,
```

Comment:
Why is this different? https://github.com/rancher/csp-adapter/blob/d8ef954b0e111203872bec9cf5dc1c5e09d59c93/pkg/supportconfig/generator.go#L31

Reply:
The requirements for this value is that `SupportEligible` is missing from the supportconfig bundle in the standard case. If the csp-adapter is installed, the value should be present and set to true. To support this requirement, I set the supportconfig struct here to have an `omitempty` tag for this value and defaulted it to false (which omits it from the output). Because we merge values from the CSP adapter, the adapter will produce a configmap with this value set to true (making the value true when that value is installed).

The other repo is for the csp adapter. Meaning that the endpoint used to be present only when the adapter was installed. Now that this will be present on any rancher install, this PR needs to be able to handle producing supportconfigs for rancher installs that have the CSP adapter or don't have the csp adapter.

Reply:
Is the supportconfig actually meaningful if the csp adapter isn't installed? Could this be simplifed by just returning a 404 if the csp adapter config doesn't exist?

Reply:
Okay :+1: